hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AInputStream.java

```diff
@@ -431,8 +432,9 @@ public synchronized int read() throws IOException {
   private void onReadFailure(IOException ioe, int length, boolean forceAbort)
           throws IOException {
 
-    LOG.info("Got exception while trying to read from stream {}" +
-        " trying to recover: " + ioe, uri);
+    LOG.info("Got exception while trying to read from stream " + uri
+        + ", client: " + client + " object: " + object
```

Comment:
why the change? The previous one used SLF4J expansion, and deliberately only printed the IOEs message, not the stack. 
If you want the stack, add a separate log at debug with the full details, and use SL4J expansion of {} elements in the log string

Reply:
Thanks a lot for the review Steve! Will address other comments asap.

The reason I made the change here is, the symptom is intermittent, and the stack is helpful when we need to diagnose issues. I hope to catch it when it occurs. My worry is that turning on debug will introduce a lot of other debug messages to the logs. But if  exceptions happen here very often, we can move the stack printing to debug. Is that the case here though? Thanks.

Reply:
leave at debug. We collect statistics on it, so if you see it a lot in the stats you can turn on debug logging for this class only.